### PR TITLE
add ability to post a new comment under an article

### PR DESCRIPTION
**Pull request by Holquaria**

### CODE REVIEW

File: src/components/NewComment.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        .catch(() => {
+            setPostedComment('error')
+        })
+        e.target[0].value = ''
+    }
+
+    return <div><form onSubmit={(event) => {
+        postNewComment(event);
+      }} className="new-comment">
+        <input className="new-comment-body" type="text" name="body"></input>
```
Comment (popatre) [inline src/components/NewComment.jsx]: This needs to be textarea - to allow multi line comments. 
Also, its not currently a controlled component - react isnt in control of this at the moment and wouldnt know whats in the form until its submitted. You can run into rendering issues.

**Review by popatre — CHANGES_REQUESTED** [review]
Looks good so far, just a few things to add;

- Stop someone posting an empty comment
- Could you lock the button after posting so they can't accidentally click it twice, especially if slow?
- Can you add a little login button - that logs you in as tickle. Then stop them posting if not logged in?
- Make the form a controlled component

File: src/components/VoteBlock.jsx [hunk 2]
```diff
@@ -21,7 +21,6 @@ export const VoteBlock = ({votes, article_id}) => {
                 return currCount -= 1
             })
         updateArticleVotes(-1, article_id)
-        .then(() => {})
         .catch(() => {
```
Comment (Poonam-raj) [inline src/components/VoteBlock.jsx:24]: Line 13 can be removed

File: src/components/Comments.jsx [hunk 3]
```diff
@@ -32,6 +34,6 @@ export const Comments = () => {
                 <p className="comment-body">{comment.body}</p>
             </li>
```
Comment (Poonam-raj) [inline src/components/Comments.jsx]: Lines 29 - 35 definitely could be extracted into another component to stop this component from bloating up

File: src/components/NewComment.jsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  };
+
+  const retryPostComment = () => {
+      setError(false)
+      postComment(article_id, postedComment.body, 'tickle122').catch(() => {
+        setError(true)
+    });
+  }
+
+  console.log(newComment);
```
Comment (Poonam-raj) [inline src/components/NewComment.jsx]: Remove console logs before making PRs and definitely before merging!

File: src/components/NewComment.jsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      { postedComment === undefined ? null : error === true ? (
+        <div className="new-comment-card-error">
+
+          <p className="username">{postedComment.author}</p>
+          <p className="date">Date posted: Just now</p>
+          <p className="comment-body">{postedComment.body}</p>
+          <p>Something went wrong, please try again</p> <button onClick={() => {
+          retryPostComment()
+        }}>Try again</button>
+        </div>
```
Comment (Poonam-raj) [inline src/components/NewComment.jsx]: This error component could be extracted out to reduce bloating of component

File: src/components/NewComment.jsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+          postNewComment(event);
+            } 
+        }}
+        className="new-comment"
+      >
+        <textarea
+          className="new-comment-body"
+          onChange={(event) => {
+            setNewComment(event.target.value);
+          }}
```
Comment (Poonam-raj) [inline src/components/NewComment.jsx:48]: value attribute should ideally be here should be newComment if this is truly a controlled component

**Review by Poonam-raj — APPROVED** [review]
Few things I'd like you to extract to debloat the code, Jamie, but otherwise nice work! I had one comment about the form so feel free to address my comments and then merge this work 🚀

**Review by Poonam-raj — APPROVED** [review]
Nice work please merge!

**Pull request merged**